File: cpp/0199二叉树的右视图.cpp
#include <vector>
#include "mybitree.h"
#include <deque>

vector<int> levelOrder(TreeNode *root)
{
    vector<int> res;
    deque<TreeNode *> nodes;

    nodes.push_back(root);

    while (!nodes.empty())
    {
        int temp=101;

        int size = nodes.size();

        for (int i = 0; i < size; ++i)
        {
            TreeNode *crt = nodes.front();
            nodes.pop_front();
            if (!crt)
                continue;

            temp=crt->val;
            nodes.push_back(crt->left);
            nodes.push_back(crt->right);
        }
        if (temp!=101)
            res.push_back(temp);
    }

    return res;
}

int main()
{
}

File: python/main/0199二叉树的右视图.py
from binarytree import TreeNode,level_create
from collections import deque

def rightSideView(root: TreeNode) -> list:
    res=[]
    nodes=deque([root])

    while(nodes):
        size=len(nodes)
        temp=None

        for _ in range(size):
            crt=nodes.popleft()

            if(crt):
                temp=crt.val
                nodes.append(crt.left)
                nodes.append(crt.right)
        
        if(temp!=None):
            res.append(temp)
    return res

print(rightSideView(level_create([0,1,2,None,3,4,None,None,5,9,None,None,6,10,None])))